File: frontend/src/types/marketing.ts
export interface MarketingCampaign {
  id: string;
  workspace_id: string;
  name: string;
  channel: string;
  status: string;
  budget: number;
  spend: number;
  impressions: number;
  clicks: number;
  conversions: number;
  target_audience?: string | null;
  goals?: string | null;
  created_at: string;
}

export interface CampaignAnalysisItem {
  id: string;
  name: string;
  channel: string;
  status: string;
  budget: number;
  spend: number;
  ctr_percent: number;
  cvr_percent: number;
  cost_per_acquisition: number;
  is_overbudget: boolean;
  is_underperforming: boolean;
}

export interface MarketingPerformance {
  total_campaigns: number;
  total_budget: number;
  total_spend: number;
  overall_ctr: number;
  overall_cvr: number;
  overall_cac: number;
  campaigns: CampaignAnalysisItem[];
  underperforming_count: number;
  underperforming_campaigns: CampaignAnalysisItem[];
}

File: backend/app/schemas/marketing.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field


class CampaignCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: str = Field(..., min_length=1, max_length=255)
    channel: str = "SOCIAL_MEDIA"
    budget: float = Field(default=0.0, ge=0)
    target_audience: Optional[str] = None
    goals: Optional[str] = None


class CampaignUpdateMetrics(BaseModel):
    model_config = ConfigDict(extra="forbid")

    spend: Optional[float] = Field(None, ge=0)
    impressions: Optional[int] = Field(None, ge=0)
    clicks: Optional[int] = Field(None, ge=0)
    conversions: Optional[int] = Field(None, ge=0)


class CampaignModifyBudget(BaseModel):
    model_config = ConfigDict(extra="forbid")

    budget: float = Field(..., ge=0)


class CampaignResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True, extra="ignore")

    id: str
    workspace_id: str
    name: str
    channel: str
    status: str
    budget: float
    spend: float
    impressions: int
    clicks: int
    conversions: int
    target_audience: Optional[str] = None
    goals: Optional[str] = None
    created_at: datetime
